### PR TITLE
Add InvalidateMeasure to IVisualElementController

### PR DESCRIPTION
**Pull request by hartez**
### Description of Change ###

Prep InvalidateMeasure calls for removal of InternalsVisibleTo
Change InvalidateMeasure to InvalidateMeasureInternal; add InvalidateMeasure to IVisualElementController interface and call InvalidateMeasureInternal via explicit implementation

### CODE REVIEW

File: Xamarin.Forms.Core/Internals/InvalidationTrigger.cs [hunk 1]
```diff
@@ -1,9 +1,9 @@
 using System;
 
-namespace Xamarin.Forms
+namespace Xamarin.Forms.Internals
 {
 	[Flags]
-	internal enum InvalidationTrigger
+	public enum InvalidationTrigger
```
Comment (hartez) [inline Xamarin.Forms.Core/Internals/InvalidationTrigger.cs]: `InvalidationTrigger` is the parameter type for `IVisualElementController.InvalidateMeasure`, which is called from several renderers in Platform projects. Similar situation to `EvalRequested`, `Ticker`, `IDataTemplate`, etc.

Comment (kingces95) [inline Xamarin.Forms.Core/Internals/InvalidationTrigger.cs]: Yeah, I see that and that it's our convention. It's just that our story can't simply be, "If it's in Form's namespace then we have a higher backcompat bar". Not gonna loose any sleep over it but could be a point of confusion for customers.

**Comment by kingces95** [comment]
:+1:

**Comment by samhouts** [comment]
:+1:

**Pull request merged**